plugins/outputs/file/file.go

```diff
@@ -39,13 +43,36 @@ func (f *File) SetSerializer(serializer serializers.Serializer) {
 	f.serializer = serializer
 }
 
-func (f *File) Connect() error {
-	writers := []io.Writer{}
-
+func (f *File) Init() error {
+	var err error
 	if len(f.Files) == 0 {
 		f.Files = []string{"stdout"}
 	}
 
+	if f.CompressionLevel == 0 {
+		switch f.CompressionAlgorithm {
+		case "zstd":
+			f.CompressionLevel = 3
+		default:
+			f.CompressionLevel = -1
+		}
+	}
+
+	if f.CompressionAlgorithm == "" || f.CompressionAlgorithm == "identity" {
+		f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm)
+	} else {
+		f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm, internal.WithCompressionLevel(f.CompressionLevel))
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
How about setting `CompressionLevel = -1` in `init()` (lowercase) to denote default and then here do:
```suggestion
	var options []internal.EncodingOption
	if f.CompressionLevel >= 0 {
		options = append(options, internal.WithCompressionLevel(f.CompressionLevel))
	}

	f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm, options...)
	if err != nil {
		return err
	}
```